core/src/main/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContext.java

```diff
@@ -361,20 +361,23 @@ protected EventBus buildEventBus() {
   @SuppressWarnings("unchecked")
   protected Compressor<ByteBuf> buildCompressor() {
     DriverExecutionProfile defaultProfile = getConfig().getDefaultProfile();
-    if (defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION)) {
-      String name = defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION);
-      if (name.equalsIgnoreCase("lz4")) {
+    if (!defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION)) {
+      return Compressor.none();
+    }
+
+    String name = defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION);
+    switch (name.toLowerCase()) {
+      case "lz4":
```

Comment:
This is feeling more and more like a case for an enum.  There used to be such an enum in 3.x.... did we move away from that for a reason?

Reply:
Unfortunately we don't have `getEnum()` in `DriverExecutionProfile`.

Reply:
Sure, but you could just as easily create the enum instance within this method, something like:

```java
  enum ValidCompressor {
    LZ4,
    SNAPPY,
    NONE
  }

  @SuppressWarnings("unchecked")
  protected Compressor<ByteBuf> buildCompressor() {
    DriverExecutionProfile defaultProfile = getConfig().getDefaultProfile();
    if (!defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION)) {
      return Compressor.none();
    }

    String name = defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION);
    ValidCompressor compressor = ValidCompressor.valueOf(name.toUpperCase());
    switch (compressor) {
      case LZ4:
        return new Lz4Compressor(this);
      case SNAPPY:
        return new SnappyCompressor(this);
      case NONE:
        return Compressor.none();
      default:
        throw new IllegalArgumentException(
            String.format(
                "Unsupported compression algorithm '%s' (from configuration option %s)",
                name, DefaultDriverOption.PROTOCOL_COMPRESSION.getPath()));
    }
  }
```

This is closer to the ProtocolOptions.Compression enum in 3.x.  It's probably _marginally_ cleaner than relying on checks for explicit string values but I don't feel _terribly_ strongly about it.

Reply:
I was going to mention extensibility as an argument against an enum, but actually if someone adds their own compression scheme, they'll have to override `buildCompressor` anyway so they could handle a special case that is not an enum constant.

So like you I don't feel terribly strongly about it either. We wouldn't expose the enum in the public API anyway, so let's keep it like this for now.

Site note: I just noticed that Cassandra added a `ZstdCompressor` recently, creating a ticket for that.